File: LegendDataEditor/LegendDataEditor/Sqlite/Table/SQLiteEffectFrame.cpp
#include "SQLiteEffectFrame.h"
#include "Utility.h"
#include "SQLiteDataReader.h"
#include "SQLite.h"

string EffectFrameData::COL_ID = "ID";
string EffectFrameData::COL_LABEL = "Label";
string EffectFrameData::COL_DIRECTION = "Direction";
string EffectFrameData::COL_ACTION = "Action";
string EffectFrameData::COL_FRAME_COUNT = "FrameCount";
string EffectFrameData::COL_POSX = "PosX";
string EffectFrameData::COL_POSY = "PosY";

void SQLiteEffectFrame::query(int ID, txVector<EffectFrameData>& dataList)
{
	string conditionString;
	StringUtility::appendConditionInt(conditionString, EffectFrameData::COL_ID, ID, "");
	string queryStr = "SELECT * FROM " + mTableName + " WHERE " + conditionString;
	parseReader(mSQLite->executeQuery(queryStr), dataList);
}
void SQLiteEffectFrame::query(int ID, int direction, txVector<EffectFrameData>& dataList)
{
	string conditionString;
	StringUtility::appendConditionInt(conditionString, EffectFrameData::COL_ID, ID, " and ");
	StringUtility::appendConditionInt(conditionString, EffectFrameData::COL_DIRECTION, direction, "");
	string queryStr = "SELECT * FROM " + mTableName + " WHERE " + conditionString;
	parseReader(mSQLite->executeQuery(queryStr), dataList);
}
void SQLiteEffectFrame::query(int ID, int direction, const string& action, txVector<EffectFrameData>& dataList)
{
	string conditionString;
	StringUtility::appendConditionInt(conditionString, EffectFrameData::COL_ID, ID, " and ");
	StringUtility::appendConditionString(conditionString, EffectFrameData::COL_ACTION, action, " and ");
	StringUtility::appendConditionInt(conditionString, EffectFrameData::COL_DIRECTION, direction, "");
	string queryStr = "SELECT * FROM " + mTableName + " WHERE " + conditionString;
	parseReader(mSQLite->executeQuery(queryStr), dataList);
}
bool SQLiteEffectFrame::updateData(const EffectFrameData& data)
{
	string updateString;
	StringUtility::appendUpdateInt(updateString, EffectFrameData::COL_FRAME_COUNT, data.mFrameCount);
	StringUtility::appendUpdateIntArray(updateString, EffectFrameData::COL_POSX, data.mPosX);
	StringUtility::appendUpdateIntArray(updateString, EffectFrameData::COL_POSY, data.mPosY);
	StringUtility::removeLastComma(updateString);
	string conditionString;
	StringUtility::appendConditionInt(conditionString, EffectFrameData::COL_ID, data.mID, " and ");
	StringUtility::appendConditionInt(conditionString, EffectFrameData::COL_DIRECTION, data.mDirection, " and ");
	StringUtility::appendConditionString(conditionString, EffectFrameData::COL_ACTION, data.mAction, "");
	return doUpdate(updateString, conditionString);
}

bool SQLiteEffectFrame::insert(const EffectFrameData& data)
{
	string valueString;
	data.insert(valueString);
	StringUtility::removeLastComma(valueString);
	return doInsert(valueString);
}

File: LegendEditor/Assets/Scripts/Game/DataBase/SQLiteTable/SQLiteEffectFrame.cs
using Mono.Data.Sqlite;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

public class EffectFrameData : TableData
{
	public static string COL_ID = "ID";
	public static string COL_LABEL = "Label";
	public static string COL_DIRECTION = "Direction";
	public static string COL_ACTION = "Action";
	public static string COL_FRAME_COUNT = "FrameCount";
	public static string COL_POSX = "PosX";
	public static string COL_POSY = "PosY";
	public int mID;
	public string mLabel;
	public int mDirection;
	public string mAction;
	public int mFrameCount;
	public List<int> mPosX;
	public List<int> mPosY;
	public override void parse(SqliteDataReader reader)
	{
		mID = stringToInt(reader[COL_ID].ToString());
		mLabel = reader[COL_LABEL].ToString();
		mDirection = stringToInt(reader[COL_DIRECTION].ToString());
		mAction = reader[COL_ACTION].ToString();
		mFrameCount = stringToInt(reader[COL_FRAME_COUNT].ToString());
		stringToIntArray(reader[COL_POSX].ToString(), ref mPosX);
		stringToIntArray(reader[COL_POSY].ToString(), ref mPosY);
	}
}

public class SQLiteEffectFrame : SQLiteTable
{
	public SQLiteEffectFrame()
		:base("EffectFrame")
	{}
	public void query(int ID, out List<EffectFrameData> dataList)
	{
		string condition = "";
		appendConditionInt(ref condition, EffectFrameData.COL_ID, ID, "");
		parseReader(doQuery(condition), out dataList);
	}
	public void query(int ID, int direction, out List<EffectFrameData> dataList)
	{
		string condition = "";
		appendConditionInt(ref condition, EffectFrameData.COL_ID, ID, " and ");
		appendConditionInt(ref condition, EffectFrameData.COL_DIRECTION, direction, "");
		parseReader(doQuery(condition), out dataList);
	}
	public void query(int ID, int direction, string action, out List<EffectFrameData> dataList)
	{
		string condition = "";
		appendConditionInt(ref condition, EffectFrameData.COL_ID, ID, " and ");
		appendConditionInt(ref condition, EffectFrameData.COL_DIRECTION, direction, " and ");
		appendConditionString(ref condition, EffectFrameData.COL_ACTION, action, "");
		parseReader(doQuery(condition), out dataList);
	}
	public void updateData(EffectFrameData data)
	{
		string updateString = "";
		appendUpdateInt(ref updateString, EffectFrameData.COL_FRAME_COUNT, data.mFrameCount);
		appendUpdateIntArray(ref updateString, EffectFrameData.COL_POSX, data.mPosX);
		appendUpdateIntArray(ref updateString, EffectFrameData.COL_POSY, data.mPosY, true);
		string condition = "";
		appendConditionInt(ref condition, EffectFrameData.COL_ID, data.mID, " and ");
		appendConditionInt(ref condition, EffectFrameData.COL_DIRECTION, data.mDirection, " and ");
		appendConditionString(ref condition, EffectFrameData.COL_ACTION, data.mAction, "");
		doUpdate(updateString, condition);
	}
	public void insert(EffectFrameData data)
	{
		string valueString = "";
		appendValueInt(ref valueString, data.mID);
		appendValueString(ref valueString, data.mLabel);
		appendValueInt(ref valueString, data.mDirection);
		appendValueString(ref valueString, data.mAction);
		appendValueInt(ref valueString, data.mFrameCount);
		appendValueIntArray(ref valueString, data.mPosX);
		appendValueIntArray(ref valueString, data.mPosY, true);
		doInsert(valueString);
	}
};